### PR TITLE
Upgrade React version of manager

### PR DESCRIPTION
**Pull request by shogohida**
Issue: #20378 

Signed-off-by: Shogo Hida <shogo.hida@gmail.com>

## What I did
Upgraded React version of manager

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by thebuilder** [comment]
This will require a couple of other changes:
https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html

- `render` should be changed to use `root` from `createRoot`.
- A bunch of TypeScript types are breaking throughout, because `children` is no longer implicit in `React.FC` components. https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-typescript-definitions 
  - This might require upgrading Emotion?

**Comment by shogohida** [comment]
@thebuilder 
Thanks for your comment! I tried to solve what you mentioned but I couldn't find `render` or `Emotion` below `code/ui/manager`. Which files should I change?

**Comment by ndelangen** [comment]
@shogohida I'd expect changes here:
https://github.com/storybookjs/storybook/blob/d18331ff33b08eaf0a9f18ff4f7509a97cc98012/code/ui/manager/src/index.tsx#L80-L86

Or is my assumption incorrect?

**Comment by ndelangen** [comment]
@shogohida will you be making the required changes @thebuilder mentioned on this PR?

I'm very excited to get storybook's manager UI upgraded to react18! But this will need more work to get merged, and we definitely do not want this be much closer to the release-date; in fact I suspect @shilman might already find it too risky (even though it's the manager, addons could potentially break..?)

Can you tell me your plans for this PR? If you're stuck on anything please reach out to me on Discord. 🙏

**Comment by shogohida** [comment]
@ndelangen 
Thanks for your comment and sorry for keeping you guys waiting for a long time. I will try to work today and tomorrow and if I get stuck, I will reach you on Discord.

I was busy for my job and left this issue undone :bow: Sorry about that.

**Comment by ndelangen** [comment]
Please don't feel sorry, your efforts are appreciated!

My name is the same on discord, I'm usually online. just ping me.. You can also schedule a meeting with me here:
https://calendly.com/chromaui

**Comment by shogohida** [comment]
@ndelangen 
I made changes. Is my understanding correct?

https://github.com/storybookjs/storybook/pull/20420/commits/9dcf29ad26dbad0257dad5e74e0628beae5e745e

File: node_modules/.package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "storybook-1",
+  "lockfileVersion": 2,
+  "requires": true,
+  "packages": {}
+}
```
Comment (ndelangen) [inline node_modules/.package-lock.json:6]: please remove these, we do not use npm-cli in this repo

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "storybook-1",
+  "lockfileVersion": 2,
+  "requires": true,
+  "packages": {}
+}
```
Comment (ndelangen) [inline package-lock.json:6]: Also remove please

File: code/ui/manager/src/index.tsx [hunk 3]
```diff
@@ -82,7 +82,8 @@ export function renderStorybookUI(domNode: HTMLElement, provider: Provider) {
     throw new Error('provider is not extended from the base Provider');
   }
 
-  ReactDOM.render(<Root key="root" provider={provider} />, domNode);
+  const root = createRoot(domNode!);
+  root.render(<Root key="root" provider={provider} />);
```
Comment (ndelangen) [inline code/ui/manager/src/index.tsx:86]: Yeah I suppose this is correct, Let's see if it works 😅

**Review by ndelangen — DISMISSED** [review]

**Comment by ndelangen** [comment]
Related #20402 #17215 #20466

File: node_modules/.package-lock.json (hunk 1 above)
Comment (shogohida) [reply node_modules/.package-lock.json]: Thanks for your review! I removed this file now.

https://github.com/storybookjs/storybook/pull/20420/commits/25b128123c92c7d410234a4c1d6b02f20d4bf1ef

File: package-lock.json (hunk 2 above)
Comment (shogohida) [reply package-lock.json]: Thanks for your review! I removed this file now.

https://github.com/storybookjs/storybook/pull/20420/commits/25b128123c92c7d410234a4c1d6b02f20d4bf1ef

File: code/ui/manager/src/index.tsx (hunk 3 above)
Comment (shogohida) [reply code/ui/manager/src/index.tsx:86]: I hope it works! Thanks for your comment!

**Comment by shogohida** [comment]
@ndelangen 
Should I recover npm lockfiles? I ask this because you dismissed your review.

In addition, I checked failing tests but I don't know how to solve them....

**Comment by ndelangen** [comment]
Nope the npm lock files were removed, so I deleted the negative review. Perhaps we can ask for help from other contributors?

**Comment by shogohida** [comment]
Sure! I will ask support on Discord

**Comment by shogohida** [comment]
I read error messages of failing tests and looked for solutions on Google. I write down notes here from my investigations.

Notes:
- Failing tests of [e2e-sandboxes](https://app.circleci.com/jobs/github/storybookjs/storybook/469381) 
  - The error message is `locator.waitFor: Target closed`
  - It seems they are related to [a bug or something of Playwright](https://github.com/microsoft/playwright/issues/13090) so I'm not sure if the failing tests of e2e-sandboxes were caused by the changes of this PR. 
- Failing tests of [cra-bench](https://app.circleci.com/pipelines/github/storybookjs/storybook/38420/workflows/43cfda21-ddee-4c12-b375-75c5c59ad4e1/jobs/469367) and [react-vite-bench](https://app.circleci.com/pipelines/github/storybookjs/storybook/38420/workflows/43cfda21-ddee-4c12-b375-75c5c59ad4e1/jobs/469372)
  - The error message is `Too long with no output (exceeded 10m0s): context deadline exceeded`
  - This is related to CircleCI settings according to [this article](https://support.circleci.com/hc/en-us/articles/360045268074-Build-Fails-with-Too-long-with-no-output-exceeded-10m0s-context-deadline-exceeded-?utm_source=google&utm_medium=sem&utm_campaign=sem-google-dg--japac-en-dsa-maxConv-auth-nb&utm_term=g_-_c__dsa_&utm_content=&gclid=Cj0KCQiAiJSeBhCCARIsAHnAzT-_ksoO7LXatX15C6YaEouvKJHX8GMs21NuYVrpjC9RJCjpU7F3psUaAlbCEALw_wcB)
  - I'm not sure if this PR caused test failures.
- Failing tests of [storybook-ui](https://www.chromatic.com/build?appId=635781f3500dd2c49e189caf&number=2054)
  -  The error message is `Error: Evaluation failed: Object at ExecutionContext._evaluateInternal (/var/task/capture-v4/build/node_modules/puppeteer-core/lib/cjs/puppeteer/common/ExecutionContext.js:218:19`
  - They are related to Puppeteer
  - I'm not sure about this too

**Comment by ndelangen** [comment]
If you start storybook on this branch, it simply doesn't render anything at all:
https://635781f3500dd2c49e189caf-ltllzjqyji.chromatic.com/

**Comment by valentinpalkovic** [comment]
Closing in favour of https://github.com/storybookjs/storybook/pull/21673

**Pull request closed without merging**